zilliqa/src/consensus.rs

```diff
@@ -108,6 +111,64 @@ impl From<Hash> for MissingBlockError {
     }
 }
 
+#[derive(Debug, Clone)]
+struct TxnOrder {
+    pub nonce: u64,
+    pub gas_price: u64,
+    pub gas_limit: u64,
+    pub hash: Hash,
+    pub retries: Arc<Mutex<u64>>, // so we can modify the element in a heap
```

Comment:
The reason for the arc mutex was because I was iterating over the heap and increasing retries, the iterator is not mutable.

However, I later realised that iterating over the heap isn't actually ordered, so it has to be a pop, therefore it doesn't actually need arc mutex or AtomicU64 - my latest commit switches it to u64.